### DISCUSSION TITLE
What is the meaning of usedBy=0?

### DISCUSSION THREAD

**Discussion by perzonaljezus** [DependencyTrack/dependency-track#937] [Q&A] [answered]
The components report shows entries with usedBy = 0.
I assume it means that the component isn't in the portfolio codebase. 
E.g. in the screenshot FastInfoset version 1.2.13 is not in the code base (usedBy=0), but 1.2.16 is (usedBy=2).
![grafik](https://user-images.githubusercontent.com/5160191/107774837-d5119a80-6d3f-11eb-96f1-82ecb5a9be9b.png)

But what is the meaning of the entry FastInfoset 1.2.13?
Why is this reported, since FastInfoset 1.2.13 is not in the codebase?

I am using Dependency-Track  v3.8.0.

**Comment by nscuro** [comment, marked as answer]
Dependency-Track < 4.0.0 did not remove unreferenced components from the portfolio. You're seeing FastInfoset 1.2.13, because it was at some point a dependency of one of your projects. You will not see this behavior in DT 4.0.0 and above.
